src/nomad_material_processing/vapor_deposition.py

```diff
@@ -0,0 +1,417 @@
+        type=ThinFilm,
+    )
+
+
+    # layers = SubSection(
+    #     description="""
+    #     An ordered list (starting at the substrate) of the thin films making up the
+    #     thin film stacks.
+    #     """,
+    #     section_def=ThinFilmReference,
+    #     repeats=True,
+    # )
+
+    temperature = SubSection(
+        section_def=SubstrateTemperature,
+    )
+    heater = Quantity(
+        type=MEnum(
+            "No heating",
+            "Halogen lamp",
+            "Filament",
+            "Resistive element",
+            "CO2 laser",
+        )
+    )
+    distance_to_source = Quantity(
+        type=float,
+        unit="meter",
+        description="""
+        The distance between the substrate and the source.
+        It is an arry because multiple sources can be used.
+        """,
+        shape=["*"],
+    )
+
+    substrate_specimen = SubSection(
+        description="""
+        The thin film stack that is being evaporated on.
+        """,
+        section_def=ThinFilmStackReference, 
+        #repeats=True, 
+    )
+
+
+class Pressure(ArchiveSection):
+    m_def = Section(
+        a_plot=dict(
+            x="process_time",
+            y="pressure",
+        ),
+    )
+    pressure = Quantity(
+        type=float,
+        unit="pascal",
+        shape=["*"],
+    )
+    process_time = Quantity(
+        type=float,
+        unit="second",
+        shape=["*"],
+    )
+
+
+class GasFlow(ArchiveSection):
+    gas = SubSection(
+        section_def=PureSubstanceSection,
+    )
+    flow = Quantity(
+        type=float,
+        unit="meter ** 3 / second",
+        shape=["*"],
+    )
+    process_time = Quantity(
+        type=float,
+        unit="second",
+        shape=["*"],  #################### why is this an array ?
+    )
+
+
+class ChamberEnvironment(ArchiveSection):
+    m_def = Section(
+        a_plot=dict(
+            x="pressure/process_time",
+            y="pressure/pressure",
+        ),
+    )
+    gas_flow = SubSection(
+        section_def=GasFlow,
+        repeats=True,
+    )
+    pressure = SubSection(
+        section_def=Pressure,
+    )
+
+
+class VaporDepositionStep(ActivityStep):
+    """
+    A step of any vapor deposition process.
+    """
+
+    m_def = Section()
+    creates_new_thin_film = Quantity(  ########## let's discuss together: how should the normalization work when we reprocess
+        type=bool,
+        description="""
+        Whether or not this step creates a new thin film.
+        """,
+        default=False,
+        a_eln=ELNAnnotation(
+            component="BoolEditQuantity",
+        ),
+    )
+    duration = Quantity(
+        type=float, 
+        unit="second"
+    )
+    sources = SubSection(
+        section_def=VaporDepositionSource,
+        repeats=True,
+    )
+    substrate = SubSection(
+        section_def=Substrate,  
+        #repeats=True, ############################### why is it an array? let's make the substrate inside an array or??
```

Comment:
As we discussed this should still be repeating.

```suggestion
        repeats=True,
```

Reply:
yes